### PR TITLE
RMB-450: Updated Documentation for Cross Table Query

### PR DESCRIPTION
**Pull request by evaluk**
Updated docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1082,18 +1082,24 @@ Correct number matching must result in 3.4 == 3.400 == 0.34e1 and correct number
 If the search term is a number then a numeric mode is used for "==", "<>", "<", "<=", ">", and ">=" if the actual JSONB type of the stored value is `number`
 (JSONB has no `integer` type).
 
-### CQL2PgJSON: Cross index searches
+### CQL2PgJSON: Cross table index queries
 
-Limited cross table searches are supported.  If you desire a join across tables the following conditions must be met:
+CQL2PgJSON supports cross table joins via subquery.  This allows arbitrary depth relationships in parent->child and child->parent relationships.  Foreign keys must be uni-directional(parent->child and child->parent).
```
Comment (julianladisch) [inline README.md]: Foreign keys must be uni-directional, the direction is always from the foreign key field in the child table to the primary key id field of the parent table.

File: README.md [hunk 2]
```diff
@@ -1082,18 +1082,24 @@ Correct number matching must result in 3.4 == 3.400 == 0.34e1 and correct number
...
 
-* The join desired index must be only 1 table deep
-  - e.g.  table1 -> table2 not table1 -> table2 -> table3
+Example relationship:
+```
+table1 -> table2 -> table3
+```
+or
+```
+table1 <- table2 <- table3
```
Comment (julianladisch) [inline README.md]:
```
table1 -> table2 -> table3
```
or
```
table3 <- table2 <- table1
```

File: README.md [hunk 3]
```diff
@@ -1116,10 +1122,60 @@ Limited cross table searches are supported.  If you desire a join across tables
           "targetTable": "table1"
         }
       ]
+    },
+    {
+      "tableName": "table2",
```
Comment (julianladisch) [inline README.md]: This is the second "table2" section, there cannot be a duplicate.

File: README.md [hunk 4]
```diff
@@ -1116,10 +1122,60 @@ Limited cross table searches are supported.  If you desire a join across tables
...
+          "fieldName": "table2Id",
+          "targetTable": "table2"
+        }
+      ]
     }
   ]
 }
 ```
+## Cross Table query ambiguity
+There will be some cases where two foreign keys refer to the same table.  This ambiguity has the possibility of leading the join logic down a dead end if the foreign key does not lead to the intended table.  To resolve this there are new properties:
```
Comment (julianladisch) [inline README.md]: The second and third sentence should be reduced to:

    This ambiguity is resolved by these properties:

File: README.md (hunk 3 above)
Comment (evaluk) [reply README.md]: this was meant to say table3, updated

File: README.md (hunk 1 above)
Comment (evaluk) [reply README.md]: I am not sure what should be changed here so I removed the line.

**Pull request merged**